components/brave_wallet/browser/brave_wallet_constants.cc

```diff
@@ -99,4 +99,30 @@ const base::flat_map<std::string, std::string>& GetAnkrBlockchains() {
 
   return *blockchains;
 }
+
+// See https://0x.org/docs/introduction/0x-cheat-sheet#allowanceholder-address
+const std::string GetZeroExAllowanceHolderAddress(const std::string& chain_id) {
+  // key = chain_id, value = allowance_holder_contract_address
+  static std::map<std::string, std::string> allowance_holder_addresses = {
+      {mojom::kMainnetChainId, kZeroExAllowanceHolderCancun},
+      {mojom::kArbitrumMainnetChainId, kZeroExAllowanceHolderCancun},
+      {mojom::kAvalancheMainnetChainId, kZeroExAllowanceHolderShanghai},
+      {mojom::kBaseMainnetChainId, kZeroExAllowanceHolderCancun},
+      {mojom::kBlastMainnetChainId, kZeroExAllowanceHolderCancun},
+      {mojom::kBnbSmartChainMainnetChainId, kZeroExAllowanceHolderCancun},
+      {mojom::kLineaChainId, kZeroExAllowanceHolderLondon},
+      {mojom::kOptimismMainnetChainId, kZeroExAllowanceHolderCancun},
+      {mojom::kPolygonMainnetChainId, kZeroExAllowanceHolderCancun},
+      {mojom::kScrollChainId, kZeroExAllowanceHolderShanghai}};
+
+  auto allowance_holder_address_pair =
+      allowance_holder_addresses.find(chain_id.c_str());
+
+  if (allowance_holder_address_pair == allowance_holder_addresses.end()) {
+    // not found
+    return "";
```

Comment:
let's return an empty optional instead

Reply:
[review(supermassive): return std::nullopt instead of empty string](https://github.com/brave/brave-core/pull/26231/commits/a6df80cd6cbd795bcbdb74cb41cda0f07b653839)